mission_control/src/behaviors/fixed_rudder.cpp

```diff
@@ -79,28 +79,55 @@ BT::NodeStatus MoveWithFixedRudder::behaviorRunningProcess()
 {
   if (!goalHasBeenPublished_)
   {
-    publishGoalMsg();
-    goalHasBeenPublished_ = true;
+    auto res = getInput<std::string>("behavior_time");
+    if (!res)
+    {
+      ROS_ERROR_STREAM("error reading port [behavior_time]:" << res.error());
+      setStatus(BT::NodeStatus::FAILURE);
```

Comment:
@patapufate meta: why not throwing on construction?

Reply:
Because:
1) if I put the state = FAILURE in the constructor the mission is gonna tick in a state which has not been defined (in behavior.h). 
2) if I throw an exception the "Action" won't return the failure state, therefore the mission won't change its state to ABORTING.

Reply:
> if I throw an exception the "Action" won't return the failure state, therefore the mission won't change its state to ABORTING.

Hmm, the mission won't even build. Or will it?

Reply:
it will. I'm throwing a `BT::RuntimeError`. 
The [library](https://www.behaviortree.dev/tutorial_02_basic_ports) says that is better to read ports on the tick event instead of the constructor. It could be that the object is constructed before we put the value on the port `setOutput(port, value)`

Reply:
> It could be that the object is constructed before we put the value on the port

Hmm, that's a good point. I think we need to strike a balance between generality and user experience. It'd be best to know that a mission is ill-formed when you load it, not when you're half-way through it.